### ISSUE TITLE
Knob changes clear actions log

### ISSUE THREAD

**Issue by Alphy11** [storybookjs/storybook#10372]
**Describe the bug**
When changing a knob, the action log gets cleared because it is a "new story" to the action log.  Changing a knob should not clear the log.

**To Reproduce**
```tsx
import React, { useEffect } from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, boolean, text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

const onEffect = action('changeyMaBobby-changed-to');
const onEffectCleanup = action('changeyMaBobby-changed-from');

storiesOf('examples/hooks/useEffect', module)
    .addDecorator(withKnobs)
    .add('storyName', () => {
        function Example({ changeyMaBobby }) {
            useEffect(() => {
                onEffect(changeyMaBobby);
                return () => onEffectCleanup(changeyMaBobby);
            }, [changeyMaBobby]);

            return (
                <>
                    <button onClick={action('button')}>{changeyMaBobby}</button>
                </>
            );
        }

        return (
            <Example
                changeyMaBobby={text(
                    'Dependent value',
                    'Change me!  See what happens'
                )}
            />
        );
    });
```

press the button a couple times to generate some action logs, then change the knob, the action log will be cleared.  Adding `clearOnStoryChange: false` lets them persist.  But then they will persist on new stories. 

**Expected behavior**
The action log should persist across knob changes, as it is the same story.

**Screenshots**
N/A

**Code snippets**
Added above

**System:**
Please paste the results of `npx -p @storybook/cli@next sb info` here.
Environment Info:

  System:
    OS: Windows 10 10.0.18363
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
  Binaries:
    Node: 13.12.0 - C:\Program Files\nodejs\node.EXE
    npm: 6.14.4 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: 44.18362.449.0

**Comment by shilman** [comment]
Hi gang, We’ve just released addon-controls in 6.0-beta!

Controls are portable, auto-generated knobs that are intended to replace addon-knobs long term.

- [Controls README](https://github.com/storybookjs/storybook/tree/next/addons/controls)
- [Migration instructions](https://github.com/storybookjs/storybook/blob/next/addons/controls/README.md#how-do-i-migrate-from-addon-knobs)

Please upgrade and try them out today. Thanks for your help and support getting this stable for release!

**Comment by shilman** [comment]
For anybody who is interested in Controls but don't know where to start, I've created a quick & dirty step-by-step walkthrough to go from a fresh CRA project to a working demo. Check it out:

=> [Storybook Controls w/ CRA & TypeScript](https://gist.github.com/shilman/69c1dd41a466bae137cc88bd2c6ef487)

There are also some "knobs to controls" migration docs in the Controls README:

=> [How do I migrate from addon-knobs?](https://github.com/storybookjs/storybook/blob/next/addons/controls/README.md#how-do-i-migrate-from-addon-knobs)
